interfaces/builtin/userns.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const userNSSummary = `allows the ability to use user namespaces`
+
+// This interface is super-privileged
+const userNSBaseDeclarationSlots = `
+  userns:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const userNSConnectedPlugAppArmorSupported = `
+# Description: allow to use user namespaces
+
+userns,
+`
+
+const userNSConnectedPlugSeccomp = `
+# allow the use of clone and unshare to use new user namespaces
+clone
+unshare
+`
+
+type userNSInterface struct {
+	commonInterface
+}
+
+func (iface *userNSInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              userNSSummary,
+		BaseDeclarationSlots: userNSBaseDeclarationSlots,
+	}
+}
+
+func (iface *userNSInterface) Name() string {
+	return "userns"
+}
+
+func (iface *userNSInterface) checkUserNSAppArmorSupport() error {
```

Comment:
If we don't actually care about the error, then shouldn't this just return a bool?

Reply:
Good point - I reworked this in 3c46203038 - let me know if you think this ok.